src/location/location.service.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpService } from '@nestjs/axios';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { firstValueFrom } from 'rxjs';
+import { ErrorMessages } from 'src/main/constants/messages.constants';
+
+@Injectable()
+export class LocationService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async getCountries() {
+    try {
+      const response = await firstValueFrom(
```

Comment:
Do you know what it do `firstValueFrom` ?

Reply:
@VYaniv, the firstValueFrom method converts an Observable to a Promise. If I understand correctly, it allows us to work with the response as a Promise and use async/await workflows.